lib/record_store/zone.rb

```diff
@@ -106,6 +133,10 @@ def provider
       Provider.const_get(config.provider).new(zone: name.gsub(/\.\z/, ''))
     end
 
+    def write
+      self.class.write(name, config: config.to_hash, records: records.map(&:to_hash))
```

Comment:
When will write be used without an instance of `Zone`? In other words, what's the reasoning behind extracting the values of a zone and using a static method of `write` instead of placing all of the logic here?